src/components/Filter/Filter.js

```diff
@@ -1,75 +1,21 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import TextField from '@material-ui/core/TextField'
-import InputAdornment from '@material-ui/core/InputAdornment'
-import IconButton from '@material-ui/core/IconButton'
-import Search from '@material-ui/icons/Search'
-import Close from '@material-ui/icons/Close'
-import { withStyles } from '@material-ui/core/styles'
-
-import { styles } from './styles/Filter.style'
-
-let ref = null
-
-const onChangeWrapper = (onChange, onClear, event) => {
-    event.target.value.length ? onChange(event.target.value) : onClear()
-}
-
-const onKeyDownWrapper = (onClear, event, text) => {
-    if (event.key === 'Escape') {
-        event.preventDefault()
-        !text.length && ref ? ref.blur() : onClear()
-    }
-}
-
-export const Filter = props => (
-    <TextField
-        autoFocus={props.autoFocus}
-        style={props.style}
-        className={props.classes.inputContainer}
-        placeholder={props.placeholder}
-        value={props.text}
-        onChange={e => onChangeWrapper(props.onChange, props.onClear, e)}
-        onKeyDown={e => onKeyDownWrapper(props.onClear, e, props.text)}
-        inputRef={node => (ref = node)}
-        InputProps={{
-            disableUnderline: props.disableUnderline,
-            style: styles.placeholder,
-            startAdornment: (
-                <InputAdornment>
-                    <Search style={styles.searchIcon} />
-                </InputAdornment>
-            ),
-            endAdornment: props.text.length ? (
-                <InputAdornment>
-                    <IconButton
-                        style={styles.iconButton}
-                        onClick={props.onClear}
-                        disableRipple
-                    >
-                        <Close style={styles.closeIcon} />
-                    </IconButton>
-                </InputAdornment>
-            ) : null,
-        }}
+import { InputField } from '@dhis2/ui-core'
+
+export const Filter = ({ text, onChange, onClear, placeholder }) => (
```

Comment:
Are we removing the search and close icons and the ability to clear the text?

Reply:
Yes, ui-core supports the native in-browser `x` instead. The icon was removed as this will be redesigned soon anyway, so didn't spend too much time to "recreate" the old design with the new components. Verified by @cooper-joe